File: postgres_39/struct/Area.go
package dto

import (
	"time"
)

type Area struct {
	Area_id               string     `json:"area_id"`
	Area_type             string     `json:"area_type"`
	Active_ind            *string    `json:"active_ind"`
	Area_max_latitude     *float64   `json:"area_max_latitude"`
	Area_max_longitude    *float64   `json:"area_max_longitude"`
	Area_min_latitude     *float64   `json:"area_min_latitude"`
	Area_min_longitude    *float64   `json:"area_min_longitude"`
	Coord_acquisition_id  *string    `json:"coord_acquisition_id"`
	Coord_system_id       *string    `json:"coord_system_id"`
	Effective_date        *time.Time `json:"effective_date"`
	Expiry_date           *time.Time `json:"expiry_date"`
	Local_coord_system_id *string    `json:"local_coord_system_id"`
	Ppdm_guid             string     `json:"ppdm_guid"`
	Preferred_name        *string    `json:"preferred_name"`
	Remark                *string    `json:"remark"`
	Source                *string    `json:"source"`
	Source_document_id    *string    `json:"source_document_id"`
	Row_changed_by        *string    `json:"row_changed_by"`
	Row_changed_date      *time.Time `json:"row_changed_date"`
	Row_created_by        *string    `json:"row_created_by"`
	Row_created_date      *time.Time `json:"row_created_date"`
	Row_effective_date    *time.Time `json:"row_effective_date"`
	Row_expiry_date       *time.Time `json:"row_expiry_date"`
	Row_quality           *string    `json:"row_quality"`
}

File: create_tables/area.py
import oracledb
import sys
import lib_platform as platform

if platform.system == "darwin":
    sys.path.insert(0, "database")
elif platform.system == "windows":
    sys.path.insert(0, "../database")

from conn import connection

# pw = getpass.getpass("Enter password: ")

# connection = oracledb.connect(
#     user="admin",
#     password="ppdm-3.9",
#     dsn="/kei-oracle-test.chkxr21l2l5s.ap-southeast-3.rds.amazonaws.com:1521/ORCL")

# print("Successfully connected to Oracle Database")

cursor = connection.cursor()


cursor.execute("""
    begin
        execute immediate 'drop table area';
        exception when others then if sqlcode <> -942 then raise; end if;
    end;""")

cursor.execute(
    """create table area(
    AREA_ID VARCHAR(40) PRIMARY KEY NOT NULL,
    AREA_TYPE VARCHAR(40) NOT NULL,
    ACTIVE_IND VARCHAR(1),
    AREA_MAX_LATITUDE DECIMAL(14,9),
    AREA_MAX_LONGITUDE DECIMAL(14,9),
    AREA_MIN_LATITUDE DECIMAL(14,9),
    AREA_MIN_LONGITUDE DECIMAL(14,9),
    COORD_ACQUISITION_ID VARCHAR(40),
    COORD_SYSTEM_ID VARCHAR(40),
    EFFECTIVE_DATE TIMESTAMP(0),
    EXPIRY_DATE TIMESTAMP(0),
    LOCAL_COORD_SYSTEM_ID VARCHAR(40),
    PPDM_GUID VARCHAR(38),
    PREFERRED_NAME VARCHAR(255),
    REMARK VARCHAR(240),
    SOURCE VARCHAR(40),
    SOURCE_DOCUMENT_ID VARCHAR(40),
    ROW_CHANGED_BY VARCHAR(30),
    ROW_CHANGED_DATE TIMESTAMP(0),
    ROW_CREATED_BY VARCHAR(30),
    ROW_CREATED_DATE TIMESTAMP(0),
    ROW_EFFECTIVE_DATE TIMESTAMP(0),
    ROW_EXPIRY_DATE TIMESTAMP(0),
    ROW_QUALITY VARCHAR(40))""")